### ISSUE TITLE
Feature Request: New performance counter for total bytes allocated

### ISSUE THREAD

**Issue by NickCraver** [dotnet/runtime#62498]
In monitoring production and in benchmark runs, we often want to see how much we're allocating. A performance counter exists for this today as "Allocated Bytes/second" ([ref link](https://docs.microsoft.com/en-us/dotnet/framework/debug-trace-profile/performance-counters)).  The problem with this _it's already a rate_. What we need for many purposes is a total that you can divide out over other things - the most common pattern for performance counters. For example, allocations per minute or hour, or allocations per request when running benchmarks.

Context:
- A rate isn't usable in aggregate over time - since the update period to an external observer is neither consistent or deterministic ("when a GC runs", we don't know what time period it's over and now can we guarantee we observed every change of the counter, meaning we could under counter. If we observe the counter once every `n` `<unit>`s, we equally miss and over or under report based of recording the same value multiple times or missing others.
- There is _a_ way to get this by listening to GC trace today, but I'd argue that's .NET-monitoring specific and not nearly as usable. Performance counters have a rich ecosystem of observers agnostic to the process type and don't a) require the permissions to listen to the process and b) already work/collect with existing infrastructure without as much overhead. I'd also argue it's not trivial to get startup allocations or collections with GC Trace.

Suggestions:
- IMO, it doesn't need to be realtime as that would incur a lot of tracking overhead. At the same as when we update existing GC counters (when a collection runs), I propose we increment this new counter.
- To match existing perf counter naming (not just in .NET), I propose this be called `Total Allocated Bytes`
- I don't think we should look at this feature request in isolation - there are other similar cases like "% Time in GC" that if we had a running total of milliseconds or something would be handy if they make sense and are in the same work area. IMO, it's prudent to see if there are a few sibling counters that make sense as additions here.

cc @Maoni0 @sebastienros

**Comment by Maoni0** [comment]
yes, we can add the total alloc bytes and the total pause time each time we do a GC. I also had a feature request to add the total pause time in `GC.GetGCMemoryInfo`, I think adding the running total of allocated bytes would be helpful there too.

**Comment by Maoni0** [comment]
also CC @cshung

**Comment by stephentoub** [comment]
The System.Runtime event source already exposes this:
https://github.com/dotnet/runtime/blob/12a8819eee9865eb38bca6c05fdece1053102854/src/libraries/System.Private.CoreLib/src/System/Diagnostics/Tracing/RuntimeEventSource.cs#L88
While that's displayed as a rate by dotnet-counters, I was under the impression that's actually dotnet-counters doing the math and the raw event that's published is actually the total bytes allocated.  Is that not the case?

**Comment by stephentoub** [comment]
> Is that not the case?

It's not the case :)  Just tried it.  The actual payload includes the rate.  Nothing to see here...

**Comment by Maoni0** [comment]
I don't think we have time for this 7.0. moving this out; let me know if you disagree.
